### PR TITLE
feat: upgraded the app with scheduler module to SDK v0.50

### PR DESCRIPTION
**Pull request by deepan95dev**

### CODE REVIEW

File: x/scheduler/keeper/wasm_handler_test.go
Comment (byte-bandit) [inline x/scheduler/keeper/wasm_handler_test.go:3]: This will come back later?

File: x/scheduler/module.go
Comment (byte-bandit) [inline x/scheduler/module.go:103]: When is this coming?

File: x/scheduler/types/keeper.go
Comment (byte-bandit) [inline x/scheduler/types/keeper.go:16]: Some of the keeper functions are using `sdk.Context`, now some others are using `context.Context`. Where's the difference coming from and can we align on just using one of them?

**Review by byte-bandit — CHANGES_REQUESTED** [review]

File: x/scheduler/keeper/wasm_handler_test.go
Comment (deepan95dev) [reply x/scheduler/keeper/wasm_handler_test.go:3]: yes, test files require some of the other modules. After the merging the modular PRs, will enable these.

File: x/scheduler/module.go
Comment (deepan95dev) [reply x/scheduler/module.go:103]: There is no usage for this function but required by sdk keeper struct.

[Ref](https://github.com/cosmos/cosmos-sdk/blob/7d5c2dbae68530bc9070983ce0b2c8a1efe23412/x/bank/module.go#L100) usage in sdk bank module

File: x/scheduler/types/keeper.go
Comment (deepan95dev) [reply x/scheduler/types/keeper.go:16]: We replaced the structs defined within paloma modules and types to `context.Context` we did not replace the ones which were defined by cosmos-sdk struct. [Here](https://github.com/cosmos/cosmos-sdk/blob/7d5c2dbae68530bc9070983ce0b2c8a1efe23412/x/group/keeper/keeper.go#L238) is the sample keeper from cosmos-sdk which still uses sdk.context

Comment (byte-bandit) [reply x/scheduler/types/keeper.go:16]: What's the idea behind this? Why not use `sdk.Context` for custom handlers as well?

File: x/scheduler/module.go
Comment (byte-bandit) [reply x/scheduler/module.go:103]: In that case, remove the TODO and replace with the contents of your comment, including the reference link. Thanks.

Comment (deepan95dev) [reply x/scheduler/module.go]: Done

File: x/scheduler/types/keeper.go
Comment (deepan95dev) [reply x/scheduler/types/keeper.go]: Sure, I just checked there were few ununsed types which were using `sdk.context` now updated to `context.Context`

**Comment by aleem1314** [comment]
Check #1036 PR review comments. Apply relevant changes here too.

**Comment by deepan95dev** [comment]
> Check #1036 PR review comments. Apply relevant changes here too.

done

File: proto/palomachain/paloma/scheduler/tx.proto
Comment (byte-bandit) [inline proto/palomachain/paloma/scheduler/tx.proto:31]: Do we really need this now? Paloma isn't really using params yet.

If so, please make sure you add the `Metadata` field that all other Paloma messages are embedding.

File: x/scheduler/keeper/keeper.go
Comment (byte-bandit) [inline x/scheduler/keeper/keeper.go:73]: Use `liblog` for logging.

Comment (byte-bandit) [reply x/scheduler/keeper/keeper.go:73]: Everywhere, exclusively.  No other logging calls in production code.

File: x/scheduler/keeper/msg_server.go
Comment (byte-bandit) [inline x/scheduler/keeper/msg_server.go:40]: Does the `0.50` upgrade enforce this?

File: x/scheduler/types/msgs.go
Comment (byte-bandit) [inline x/scheduler/types/msgs.go:18]: Paloma is using a field called `Metadata` for all custom messages. Please take a look at other messages to see how it's used and adapt this to match.

**Review by byte-bandit — CHANGES_REQUESTED** [review]

File: x/scheduler/keeper/keeper.go
Comment (deepan95dev) [reply x/scheduler/keeper/keeper.go:73]: Did you mean to replace the usage of fmt here? It was there in the master branch already [here](https://github.com/palomachain/paloma/blob/72fcb671899c8eb0557e5342cd7547e9655e14c5/x/scheduler/keeper/keeper.go#L75)

File: x/scheduler/keeper/msg_server.go
Comment (deepan95dev) [reply x/scheduler/keeper/msg_server.go:40]: let me check the usage of this function in the scheduler module and update

File: x/scheduler/keeper/keeper.go
Comment (byte-bandit) [reply x/scheduler/keeper/keeper.go:73]:
```suggestion
		return liblog.FromSDKLogger(ctx.Logger().With("module", fmt.Sprintf("x/%s", types.ModuleName)))
```

**Review by aleem1314 — CHANGES_REQUESTED** [review]
Still there are few things need to be updated.
Check https://github.com/palomachain/paloma/pull/1036#pullrequestreview-1763965986

File: x/scheduler/keeper/msg_server.go
Comment (deepan95dev) [reply x/scheduler/keeper/msg_server.go:40]: sorry for the delay, just verified that above functions are required and used in module.go https://github.com/palomachain/paloma/blob/469ba6bf4ecc59010d7c25ce35ce2162ff49c8ac/x/scheduler/module.go#L133

File: x/scheduler/types/msgs.go
Comment (deepan95dev) [reply x/scheduler/types/msgs.go]: Updated to default and pushed.

File: x/scheduler/keeper/keeper.go
Comment (deepan95dev) [reply x/scheduler/keeper/keeper.go]: Updated and pushed

File: proto/palomachain/paloma/scheduler/tx.proto
Comment (deepan95dev) [reply proto/palomachain/paloma/scheduler/tx.proto]: removed and pushed

File: x/scheduler/keeper/msg_server.go
Comment (deepan95dev) [reply x/scheduler/keeper/msg_server.go]: removed and pushed

File: x/scheduler/types/msgs.go
Comment (byte-bandit) [inline x/scheduler/types/msgs.go:17]: Please take a look at how this interface is implemented for other messages in Paloma and update this code to match.

File: x/scheduler/types/msgs.go
Comment (byte-bandit) [inline x/scheduler/types/msgs.go:31]: Also this.

**Review by byte-bandit — CHANGES_REQUESTED** [review]

File: app/app.go
Comment (aleem1314) [inline app/app.go:609]: EVM module is enabled. You should pass EVM module keeper here.

File: app/app.go
Comment (aleem1314) [inline app/app.go:611]:
```suggestion
			app.EvmKeeper,
```

File: x/scheduler/keeper/exported/exported.go
Comment (aleem1314) [inline x/scheduler/keeper/exported/exported.go:1]: This file is not required. You can delete this.

File: x/scheduler/keeper/keeper.go
Comment (aleem1314) [inline x/scheduler/keeper/keeper.go:27]: You should use same type for both `storeKey` and `memKey`.

File: x/scheduler/keeper/keeper.go
Comment (aleem1314) [inline x/scheduler/keeper/keeper.go:59]: Where are we using `authority`?

File: x/scheduler/keeper/keeper.go
Comment (aleem1314) [inline x/scheduler/keeper/keeper.go:85]: This is not correct. You are returning `memStore`, it should be kvstore.

File: x/scheduler/keeper/keeper.go
Comment (aleem1314) [inline x/scheduler/keeper/keeper.go:97]: Why this change?

File: x/scheduler/types/msgs.go
Comment (aleem1314) [inline x/scheduler/types/msgs.go:8]: The `schedular` module does not have any params. You can remove this file.

File: app/app.go
Comment (deepan95dev) [reply app/app.go]: enabled and pushed

File: app/app.go
Comment (deepan95dev) [reply app/app.go]: done

File: third_party/cosmos/msg/v1/msg.proto
Comment (deepan95dev) [reply third_party/cosmos/msg/v1/msg.proto]: deleted

File: x/scheduler/keeper/exported/exported.go
Comment (deepan95dev) [reply x/scheduler/keeper/exported/exported.go]: deleted

File: x/scheduler/types/msgs.go
Comment (deepan95dev) [reply x/scheduler/types/msgs.go]: deleted this file for scheduler module

File: x/scheduler/types/msgs.go
Comment (deepan95dev) [reply x/scheduler/types/msgs.go]: deleted this file for scheduler module

File: proto/palomachain/paloma/scheduler/tx.proto
Comment (byte-bandit) [inline proto/palomachain/paloma/scheduler/tx.proto:6]: Unneeded import?

File: x/scheduler/keeper/keeper.go
Comment (deepan95dev) [reply x/scheduler/keeper/keeper.go]: Updated and pushed

File: x/scheduler/keeper/keeper.go
Comment (deepan95dev) [reply x/scheduler/keeper/keeper.go]: removed and pushed

File: x/scheduler/keeper/keeper.go
Comment (deepan95dev) [reply x/scheduler/keeper/keeper.go:97]: reverted and pushed

File: x/scheduler/types/msgs.go
Comment (deepan95dev) [reply x/scheduler/types/msgs.go]: this file was deleted.

File: proto/palomachain/paloma/scheduler/tx.proto
Comment (deepan95dev) [reply proto/palomachain/paloma/scheduler/tx.proto]: updated and pushed

File: x/scheduler/types/tx.pb.go
Comment (deepan95dev) [reply x/scheduler/types/tx.pb.go]: generated the new proto files and pushed

File: x/scheduler/keeper/keeper.go
Comment (deepan95dev) [reply x/scheduler/keeper/keeper.go:85]: I pushed above fix with reference link with comment [here](https://github.com/palomachain/paloma/pull/1044/commits/7f8e981e399c3b70a0631685787434721c9dd1ae) Once verify if its correct usage @aleem1314  @byte-bandit

File: x/scheduler/types/mocks/BankKeeper.go
Comment (byte-bandit) [inline x/scheduler/types/mocks/BankKeeper.go:1]: Why are these new? Are they used anywhere?

**Review by byte-bandit — COMMENTED** [review]
LGTM. Please clean up unused files and make sure the linters run successfully. 

Revisit your commit history to stash and fixup where useful. Please use [convential commits](https://www.conventionalcommits.org/en/v1.0.0/) and make sure you [sign your commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits).

File: x/scheduler/types/mocks/BankKeeper.go
Comment (deepan95dev) [reply x/scheduler/types/mocks/BankKeeper.go]: deleted unused mock files and pushed.

**Comment by deepan95dev** [comment]
> LGTM. Please clean up unused files and make sure the linters run successfully.
> 
> Revisit your commit history to stash and fixup where useful. Please use [convential commits](https://www.conventionalcommits.org/en/v1.0.0/) and make sure you [sign your commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits).

noted. Will follow the commit conventions and verified signatures for further commits.

**Review by byte-bandit — DISMISSED** [review]
LGTM, final review to @aleem1314

File: x/scheduler/types/params_legacy.go
Comment (aleem1314) [inline x/scheduler/types/params_legacy.go:1]: You can delete this file.

**Review by aleem1314 — APPROVED** [review]
Left one comment. Pre approving

File: x/scheduler/types/params_legacy.go
Comment (deepan95dev) [reply x/scheduler/types/params_legacy.go]: done

**Comment by deepan95dev** [comment]
@byte-bandit rebase and fixup done.

File: x/scheduler/types/params_legacy.go
Comment (taariq) [reply x/scheduler/types/params_legacy.go]: Resolving.

File: x/scheduler/types/mocks/BankKeeper.go
Comment (taariq) [reply x/scheduler/types/mocks/BankKeeper.go]: Resolved.

**Review by taariq — APPROVED** [review]
@byte-bandit @verabehr approving for merge.

**Comment by taariq** [comment]
@aleem1314 requires your approval here on review.

**Review by aleem1314 — APPROVED** [review]

**Pull request merged**
